### PR TITLE
Remove VM deprecations before 3.0 release

### PR DESCRIPTION
**Pull request by vbauzysvmware**
* From VM resource removed deprecated "ip", "network_name", "vapp_network_name", "network_href", "mac", "initscript" and related code
* Fix for #541 - used code from this PR and removed not needed code anymore.

### CODE REVIEW

File: .changes/v3.0.0/563-deprecations.md
Comment (lvirbalas) [inline .changes/v3.0.0/563-deprecations.md:1]: This information needs to appear in the documentation in that section which @dataclouder added in one of the PRs about removed resources. We need to list these removals in the docs for at least one release of 3.0. We will remove it in further releases.

File: vcd/resource_vcd_vapp_vm.go
Comment (lvirbalas) [inline vcd/resource_vcd_vapp_vm.go:1661]: What will happen with #541 given the above block is now here?

**Review by lvirbalas — CHANGES_REQUESTED** [review]
Funny that feeling when you see code removed... :) Few remarks in-line.

File: .changes/v3.0.0/563-deprecations.md
Comment (vbauzysvmware) [reply .changes/v3.0.0/563-deprecations.md:1]: this removals are fields. Ar you sure?

File: vcd/resource_vcd_vapp_vm.go
Comment (vbauzysvmware) [reply vcd/resource_vcd_vapp_vm.go:1661]: closed

File: .changes/v3.0.0/563-deprecations.md
Comment (lvirbalas) [reply .changes/v3.0.0/563-deprecations.md:1]: Yes. Maybe only section header needs to change (I'm not sure what it's called right now), but we need to state all removals of 3.0 in a single doc place.

File: vcd/resource_vcd_vapp_vm.go
Comment (lvirbalas) [reply vcd/resource_vcd_vapp_vm.go:1661]: Strange, it shows open for me and pending for my review :)

Comment (vbauzysvmware) [reply vcd/resource_vcd_vapp_vm.go:1661]: will be closed, then this PR will have success merge

File: .changes/v3.0.0/563-deprecations.md
Comment (vbauzysvmware) [reply .changes/v3.0.0/563-deprecations.md:1]: added

Comment (lvirbalas) [reply .changes/v3.0.0/563-deprecations.md:1]: Thanks, careful with conflict resolution when you merge ;)

File: website/docs/r/vapp_vm.html.markdown
Comment (lvirbalas) [inline website/docs/r/vapp_vm.html.markdown:248]: When I run my configuration on this PR it results in an unexpected change:

```
  # vcd_vapp.demo_vapp will be updated in-place
  ~ resource "vcd_vapp" "demo_vapp" {
      + accept_all_eulas = true
        guest_properties = {}
        href             = "https://bos1-vcloud-static-172-8.eng.vmware.com/api/vApp/vapp-7752b19f-52e2-4fae-8a61-5297319907dd"
        id               = "urn:vcloud:vapp:7752b19f-52e2-4fae-8a61-5297319907dd"
        ip               = "allocated"
        metadata         = {}
        name             = "demo-web"
        power_on         = true
        status           = 4
        status_text      = "POWERED_ON"
    }
```

This is my config:

```
# vApp
resource "vcd_vapp" "demo_vapp" {
  name = "demo-web"

  power_on = true

  depends_on = ["vcd_network_routed.demo_routed_net_web"]
}
```

File: website/docs/r/vapp_vm.html.markdown
Comment (lvirbalas) [inline website/docs/r/vapp_vm.html.markdown:370]: We have deprecated `power_on` in the previous PR https://github.com/vmware/terraform-provider-vcd/pull/558
Won't that cause issues with the above?
CC: @Didainius

**Review by lvirbalas — CHANGES_REQUESTED** [review]

File: website/docs/r/vapp_vm.html.markdown
Comment (vbauzysvmware) [reply website/docs/r/vapp_vm.html.markdown:370]: VM have own power on handling, no issue here. Issue could be if power_on would be used together on VM and vApp.

File: website/docs/r/vapp_vm.html.markdown
Comment (vbauzysvmware) [reply website/docs/r/vapp_vm.html.markdown:248]: will be removed in https://github.com/vmware/terraform-provider-vcd/pull/558

Comment (lvirbalas) [reply website/docs/r/vapp_vm.html.markdown:248]: Looks like I got this, because was using build from #563, which removed the field, and then switched to this branch, which doesn't have the removal yet and wants to add it.

**Review by lvirbalas — APPROVED** [review]
LGTM! Just one ask: after merging this and #558 please double check the "Removed resources and fields" doc section that it includes all items as expected.

**Review by Didainius — APPROVED** [review]

**Review by dataclouder — APPROVED** [review]

**Pull request merged**
